src/llmq/quorums_instantsend.cpp

```diff
@@ -624,8 +624,6 @@ void CInstantSendManager::HandleNewInstantSendLockRecoveredSig(const llmq::CReco
         }
 
         islock = std::make_shared<CInstantSendLock>(std::move(it->second));
-        creatingInstantSendLocks.erase(it);
-        txToCreatingInstantSendLocks.erase(islock->txid);
```

Comment:
You need to move these lines into `if (islock->txid != recoveredSig.msgHash) {` block below then I think.

Reply:
Why?

Reply:
Because otherwise these conflicting entries would never be deleted I think, no?